VSS.md

```diff
@@ -0,0 +1,484 @@
+
+**Operator Precedence**
+
+Typical operator precedence rules apply, i.e., Logical negate is the highest priority, followed by Logical and and then Logical Or with the lowest priority. Parenthesis can be used to override the default precedence rules.
+
+**Examples of Combining Logical Operators**
+
+Logical operators can be combined to form complex filter expressions.
+
+The following query will return all books with "comedy" or "horror" genre (AND) published between 2015 and 2024:
+
+`@genre:[comedy|horror] @year:[2015 2024]`
+
+The following query will return all books with "comedy" or "horror" genre (OR) published between 2015 and 2024:
+
+`@genre:[comedy|horror] | @year:[2015 2024]`
+
+The following query will return all books that either don't have a genre field, or have a genre field not equal to "comedy", that are published between 2015 and 2024:
+
+`-@genre:[comedy] @year:[2015 2024]`
+
+### Configuration 
+
+1\. **reader-threads:** (Integer) Controls the amount of threads executing queries.  
+2\. **writer-threads:** (Integer) Controls the amount of threads processing index mutations.  
+3\. **use-coordinator:** (boolean) Cluster mode enabler.
+
+### Dependencies 
+
+1. [Abseil](https://github.com/abseil/abseil-cpp): A collection of C++ libraries designed to augment the standard library.  
+2. [gRPC](https://github.com/grpc/grpc): A high-performance, open-source universal RPC framework.  
+   
+
+Additionally, the codebase integrates a forked version of hnswlib.
+
+### Testing 
+
+1. **Unittesting:** code coverage currently stands at 90%.  
+2. **Stability test framework:** fuzzy testing based on python.
+
+### Observability 
+
+The following metrics are added to the INFO command.
+
+- **search\_total\_indexed\_hash\_keys**	(Integer)	Total count of HASH keys for all indexes  
+- **search\_number\_of\_indexes**	(Integer)	Index schema total count  
+- **search\_number\_of\_attributes**	(Integer)	Total count of attributes for all indexes  
+- **search\_failure\_requests\_count**	(Integer)	A count of all failed requests, including syntax errors.
+- **search\_successful\_requests\_count**	(Integer)	A count of all successful requests  
+- **search\_used\_memory\_human**	(Integer)	A human-friendly readable version of the search\_used\_memory\_bytes metric  
+- **search\_used\_memory\_bytes**	(Integer)	The total bytes of memory that all indexes occupy  
+- **search\_background\_indexing\_status**	(String)	The status of the indexing process. NO\_ACTIVITY indicates idle indexing.  
+- **search\_hnsw\_create\_exceptions\_count**	(Integer)	Count of HNSW creation exceptions.  
+- **search\_hnsw\_search\_exceptions\_count**	(Integer)	Count of HNSW search exceptions  
+- **search\_hnsw\_remove\_exceptions\_count**	(Integer)	Count of HNSW removal exceptions.  
+- **search\_hnsw\_add\_exceptions\_count**	(Integer)	Count of HNSW addition exceptions.  
+- **search\_hnsw\_modify\_exceptions\_count**	(Integer)	Count of HNSW modification exceptions  
```

Comment:
Still unclear, what are exceptions? How do these fails and what are end users supposed to be doing about these. Are these just syntax errors?